helix-term/src/ui/mod.rs

```diff
@@ -117,18 +115,25 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
         // in our picker.
         .filter_entry(|entry| entry.file_name() != ".git");
 
-    let walk_builder = match type_builder.add(
-        "compressed",
-        "*.{zip,gz,bz2,zst,lzo,sz,tgz,tbz2,lz,lz4,lzma,lzo,z,Z,xz,7z,rar,cab}",
-    ) {
-        Err(_) => &walk_builder,
-        _ => {
-            type_builder.negate("all");
-            let excluded_types = type_builder.build().unwrap();
-            walk_builder.types(excluded_types)
-        }
-    };
+    // We want to exclude files that the editor can't handle yet
+    let mut type_builder = TypesBuilder::new();
+
+    type_builder
+        .add(
+            "compressed",
+            "*.{zip,gz,bz2,zst,lzo,sz,tgz,tbz2,lz,lz4,lzma,lzo,z,Z,xz,7z,rar,cab}",
+        )
+        // This shouldn't panic as the above is static, but if it ever
+        // is changed and becomes invalid it will catch here rather than
+        // being unnoticed.
+        .expect("Invalid type definition");
+    type_builder.negate("all");
+
+    if let Ok(excluded_types) = type_builder.build() {
```

Comment:
The `unwrap()`/`expect()` here would also seem reasonable to me since we know that the list of types specified is valid and will build. `if let Ok()` will hide the issue if this breaks one day.

I'll fix in a followup